### PR TITLE
remove unneeded item from CYA page if already removed from workflow.

### PR DESCRIPTION
**Pull request by OgunyemiO**
### Jira link (if applicable)

https://tools.hmcts.net/jira/browse/EXUI-1077

### Change description ###

remove unneeded item from CYA page if already removed from workflow.

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/hearings/containers/request-hearing/hearing-create-edit-summary/hearing-create-edit-summary.component.ts [hunk 1]
```diff
@@ -1,28 +1,62 @@
...
+        return;
+      } else {
+        const hearing = str.hearings;
+        workFlow = hearing.hearingValues.serviceHearingValuesModel.screenFlow
+      }
+    });
+    return workFlow;
+  }
+
+  removeUnnecessarySummaryTemplateItems() {
```
Comment (RiteshHMCTS) [inline src/hearings/containers/request-hearing/hearing-create-edit-summary/hearing-create-edit-summary.component.ts:46]: can you put the function type whether private/public

File: src/hearings/containers/request-hearing/hearing-create-edit-summary/hearing-create-edit-summary.component.ts [hunk 2]
```diff
@@ -1,28 +1,62 @@
...
 
+  ngOnInit(): void {
+    this.template = this.removeUnnecessarySummaryTemplateItems();
+  }
+
   public executeAction(action: ACTION): void {
     super.navigateAction(action);
   }
 
+  private getScreenFlowFromStore(hearingStore) {
```
Comment (RiteshHMCTS) [inline src/hearings/containers/request-hearing/hearing-create-edit-summary/hearing-create-edit-summary.component.ts:33]: add unit test to cover this function

File: src/hearings/containers/request-hearing/hearing-create-edit-summary/hearing-create-edit-summary.component.ts [hunk 3]
```diff
@@ -1,28 +1,62 @@
...
+import { Component, OnDestroy, OnInit } from '@angular/core';
 import { Store } from '@ngrx/store';
 import { ACTION, Mode } from '../../../models/hearings.enum';
 import { HearingsService } from '../../../services/hearings.service';
 import * as fromHearingStore from '../../../store';
 import { HEARING_CREATE_EDIT_SUMMARY_TEMPLATE } from '../../../templates/hearing-create-edit-summary.template';
 import { RequestHearingPageFlow } from '../request-hearing.page.flow';
+import { string } from '@pact-foundation/pact/src/dsl/matchers';
+import { Section } from 'src/hearings/models/section';
+import { ScreenNavigationModel } from 'api/hearings/models/screenNavigation.model';
```
Comment (connorpgpmcelroy) [inline src/hearings/containers/request-hearing/hearing-create-edit-summary/hearing-create-edit-summary.component.ts:10]: Should not have references to src, should be '../'. Also, preferably should not be using api/ but not present in src then not as necessary

**Review by connorpgpmcelroy — CHANGES_REQUESTED** [review]
Minor change requested

**Review by RiteshHMCTS — DISMISSED** [review]

**Review by richardhenryash — APPROVED** [review]

**Review by olusegz07 — APPROVED** [review]

**Review by connorpgpmcelroy — DISMISSED** [review]
Import ordering is still not correct but minor issue. Will approve

**Review by andywilkinshmcts — CHANGES_REQUESTED** [review]
Change the function getScreenFlowFrom Store to be properly async
Handle all the screens - statically declare a mapping from screen

File: src/hearings/containers/request-hearing/hearing-create-edit-summary/hearing-create-edit-summary.component.ts [hunk 4]
```diff
@@ -1,28 +1,62 @@
...
 
+  ngOnInit(): void {
+    this.template = this.removeUnnecessarySummaryTemplateItems();
+  }
+
   public executeAction(action: ACTION): void {
     super.navigateAction(action);
   }
 
+  public getScreenFlowFromStore(hearingStore) {
```
Comment (andywilkinshmcts) [inline src/hearings/containers/request-hearing/hearing-create-edit-summary/hearing-create-edit-summary.component.ts:32]: declare return type

File: src/hearings/containers/request-hearing/hearing-create-edit-summary/hearing-create-edit-summary.component.ts [hunk 5]
```diff
@@ -1,28 +1,59 @@
...
+import { Component, OnDestroy, OnInit } from '@angular/core';
 import { Store } from '@ngrx/store';
 import { ACTION, Mode } from '../../../models/hearings.enum';
 import { HearingsService } from '../../../services/hearings.service';
 import * as fromHearingStore from '../../../store';
 import { HEARING_CREATE_EDIT_SUMMARY_TEMPLATE } from '../../../templates/hearing-create-edit-summary.template';
 import { RequestHearingPageFlow } from '../request-hearing.page.flow';
+import { Section } from '../../../../hearings/models/section';
+import { ScreenNavigationModel } from 'api/hearings/models/screenNavigation.model';
+import { Observable } from 'rxjs';
```
Comment (connorpgpmcelroy) [inline src/hearings/containers/request-hearing/hearing-create-edit-summary/hearing-create-edit-summary.component.ts:10]: Imports should be ordered '@...', 'a-z...', '../../../..', '../', ./' ideally

**Review by connorpgpmcelroy — APPROVED** [review]
Approved with minor comment

**Review by kasi-subramaniam — APPROVED** [review]

File: yarn.lock [hunk 6]
```diff
@@ -19015,7 +19015,7 @@ __metadata:
     "@angular/platform-browser-dynamic": ^15.2.9
     "@angular/router": ^15.2.9
     "@edium/fsm": ^2.1.2
-    "@hmcts/ccd-case-ui-toolkit": 7.0.10
+    "@hmcts/ccd-case-ui-toolkit": 7.0.8-compress-ls
```
Comment (andywilkinshmcts) [inline yarn.lock:19063]: This is probably not the version of toolkit you want. The live version is 7.0.10

File: src/hearings/containers/request-hearing/hearing-view-edit-summary/hearing-view-edit-summary.component.spec.ts [hunk 7]
```diff
@@ -66,6 +66,19 @@ describe('HearingViewEditSummaryComponent', () => {
       expect(component.validationErrors.length).toEqual(1);
     });
 
+    it('should call removeUnnecessarySummaryTemplateItems in ngOnInit', () => {
```
Comment (andywilkinshmcts) [inline src/hearings/containers/request-hearing/hearing-view-edit-summary/hearing-view-edit-summary.component.spec.ts:69]: I really think you need a test that actually tests the functionality.  Refactor to make the template and screen flow constructor parameters (with defaults). 
Define a mock template, and mock screen flow, and pass these to the component under test. Check that the expected items are removed and that no unexpected items are removed

**Review by andywilkinshmcts — CHANGES_REQUESTED** [review]
I think we need some more tests as suggested, and to check the toolkit version

File: package.json [hunk 8]
```diff
@@ -90,7 +90,7 @@
     "@angular/platform-browser-dynamic": "^15.2.9",
     "@angular/router": "^15.2.9",
     "@edium/fsm": "^2.1.2",
-    "@hmcts/ccd-case-ui-toolkit": "7.0.10",
+    "@hmcts/ccd-case-ui-toolkit": "7.0.8-compress-ls",
```
Comment (andywilkinshmcts) [inline package.json:93]: Don't change the toolkit version

**Review by andywilkinshmcts — APPROVED** [review]
Don't change the toolkit version - other than that v good

File: package.json (hunk 8 above)
Comment (RiteshHMCTS) [inline package.json:93]: @OgunyemiO can you revert the TL version if you have not done any toolkit changes

File: src/hearings/containers/request-hearing/hearing-create-edit-summary/hearing-create-edit-summary.component.ts (hunk 5 above)
Comment (connorpgpmcelroy) [reply src/hearings/containers/request-hearing/hearing-create-edit-summary/hearing-create-edit-summary.component.ts:11]: minor issue but still not the case

**Review by HYadavHMCTS — APPROVED** [review]

**Review by connorpgpmcelroy — APPROVED** [review]

**Pull request merged**